cinema/urls.py

```diff
@@ -1,10 +1,26 @@
-from django.urls import path
+from django.urls import path, include
+from rest_framework import routers
 
-from cinema.views import movie_list, movie_detail
+from cinema.views import (
+    GenreList,
+    GenreDetail,
+    CinemaHallViewSet,
+    ActorList,
+    ActorDetail,
+    MovieViewSet
+)
+
+
+router = routers.DefaultRouter()
+router.register("cinema_halls", CinemaHallViewSet)
+router.register("movies", MovieViewSet)
 
 urlpatterns = [
-    path("movies/", movie_list, name="movie-list"),
-    path("movies/<int:pk>/", movie_detail, name="movie-detail"),
+    path("genres/", GenreList.as_view(), name="genre-list"),
+    path("genres/<int:pk>/", GenreDetail.as_view(), name="genre-detail"),
+    path("actors/", ActorList.as_view(), name="actor_list"),
+    path("actors/<int:pk>/", ActorDetail.as_view(), name="actor_list"),
+    path("", include(router.urls)),
```

Comment:
No need to include it you do not add a base path
Do as follow: urlpatters = [list of urls] + router.urls